containerd: migrate to using buildGoPackage

vdemeester opened the pull request:
###### Motivation for this change

Part of #52469, make the `runc` derivation using `buildGoPackage` :angel: 

… and add man pages, which means `containerd` becomes a multi-output
derivation : `containerd.bin` and `containerd.man`.


cc @Mic92 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Mic92 commented on pkgs/applications/virtualization/containerd/default.nix:
```diff
@@ -14,34 +13,37 @@ stdenv.mkDerivation rec {
+      manName="$(basename "$manFile")" # "docker-build.1"
+      number="$(echo $manName | rev | cut -d'.' -f1 | rev)"
+      mkdir -p "$manRoot/man$number"
+      gzip -c "$manFile" > "$manRoot/man$number/$manName.gz"
```
I think we automatically compress manpages.

vdemeester replied:
oh :thinking: then there is some other packages that should change too

vdemeester commented:
oh this broke the docker derivation :sweat_smile: I'll fix :angel:

vdemeester committed:
containerd: migrate to using buildGoPackage

… and add man pages, which means `containerd` becomes a multi-output
derivation : `containerd.bin` and `containerd.man`.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Mic92 commented:
@GrahamcOfBorg test docker

Mic92 commented:
@GrahamcOfBorg build nixosTests.docker
